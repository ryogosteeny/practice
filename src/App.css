.app {
    margin-top: 32px;
}
