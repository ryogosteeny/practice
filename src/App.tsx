import React, { VFC } from "react";
import { GlobalProvider } from "./GlobalProvider";
import { createTheme } from "@mui/material";
import { Film } from "./components/organism/Film";
import { Routes, BrowserRouter, Route } from "react-router-dom";
import { DescBox } from "./components/organism/DescBox";
import { CounterOrganism } from "./components/organism/CounterOrganism";

const theme = createTheme({
  palette: {
    primary: {
      main: "#1849a5",
    },
  },
});

export const App: VFC = () => {
  console.log("------------------------App.tsx");

  return (
    <BrowserRouter>
      <GlobalProvider>
        <Routes>
          <Route index element={<DescBox />} />
          <Route path="movies" element={<Film />} />
          <Route path="comingSoon" element={<CounterOrganism />} />
        </Routes>
      </GlobalProvider>
    </BrowserRouter>
  );
};
