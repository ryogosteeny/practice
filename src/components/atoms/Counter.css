.counter {
    margin-bottom: 8px;
    font-size: 24px;
    font-weight: 600;
}

.counter__item {
    margin-bottom: 8px;
}
