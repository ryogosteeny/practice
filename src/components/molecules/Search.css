.search-layout {
    display: flex;
    justify-content: center;
    align-items: center;
}

.search {
    margin: 24px auto;
    padding: 16px;
    max-width: 1232px;
}


.input {
    margin-right: 8px;
    padding: 6px 0;
    width: 560px;
    font-size: 1.6rem;
    border-color: transparent;
    border-radius: 4px;
}
