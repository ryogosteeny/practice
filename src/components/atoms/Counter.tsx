import React from "react";
import "./Counter.css";

interface Props {
  counterTitle: string;
  count: number;
}

export const Counter = React.memo(({ counterTitle, count }: Props) => {
  console.log("Counter :", `${counterTitle}`);
  return (
    <div className={"counter"}>
      {counterTitle}:<span className={"count"}>{count}</span>
    </div>
  );
});
