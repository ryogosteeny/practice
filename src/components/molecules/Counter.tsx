import React from "react";
import { useCounter } from "../../hooks/useCounter";
import { MainButton } from "../atoms/MainButton";
import "./Counter.css";

interface Props {
  counterTitle: string;
  buttonText: string;
}

export const Counter = ({ counterTitle, buttonText }: Props) => {
  const { count, handleCountUpCallBack } = useCounter();
  console.log("Counter :", `${counterTitle}`);
  return (
    <div className="counter">
      <div className="counter__item">
        {counterTitle}:{count}
      </div>
      <MainButton btnText={buttonText} onClick={handleCountUpCallBack} />
    </div>
  );
};
