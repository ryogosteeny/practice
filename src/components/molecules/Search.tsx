import React from "react";
import { useSearch } from "../../hooks/useSearch";
import { MainButton } from "../atoms/MainButton";
import "./Search.css";

interface Props {
  search: (searchValue: string) => void;
}

export const Search = React.memo(({ search }: Props) => {
  const { searchValue, handleSearchValue, callSearchFunction } = useSearch();
  return (
    <form className="search search-layout">
      <input
        className="input"
        type="text"
        value={searchValue}
        onChange={handleSearchValue}
      />
      <MainButton
        btnText={"検索"}
        onClick={(e: React.MouseEvent<HTMLButtonElement>) =>
          callSearchFunction(e, search)
        }
      />
    </form>
  );
});

// TODO atom単位のボタンの正しいコンポーネント設計
