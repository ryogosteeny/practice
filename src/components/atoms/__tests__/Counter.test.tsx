import React from "react";
import { render } from "@testing-library/react";
import { Counter } from "../../molecules/Counter";
import userEvent from "@testing-library/user-event";
import { App } from "../../../App";

describe("This suit is to test the Body component", () => {
  test("snapshot", () => {
    const asFragment = render(
      <Counter counterTitle={"Learn React"} count={0} />
    ).asFragment();
    expect(asFragment).toMatchSnapshot();
  });

  test("finding titleText", () => {
    const { getByText } = render(
      <Counter counterTitle={"Learn React"} count={0} />
    );
    expect(getByText("Jay Rocks")).toBeInTheDocument();
  });

  test("check Counter value after a tap", () => {
    const { getByTestId } = render(
      <Counter counterTitle={"Learn React"} count={0} />
    );
    const button = getByTestId("button");
    userEvent.click(button);
    userEvent.click(button);
    expect(getByTestId("counter")).toHaveTextContent("2");
  });

  test("App titleText", () => {
    const { getByText } = render(<App />);
    const linkElement = getByText(/ React/i);
    expect(linkElement).toBeInTheDocument();
  });
});
