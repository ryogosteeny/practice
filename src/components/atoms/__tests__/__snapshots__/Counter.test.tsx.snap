// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This suit is to test the Body component snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1
      data-testid="header"
    >
      Learn React
    </h1>
    <p
      data-testid="count"
    >
      0
    </p>
    <button
      data-testid="button"
      type="button"
    >
      Tap Me
    </button>
  </div>
</DocumentFragment>
`;
